src/main/java/teammates/ui/controller/Action.java

```diff
@@ -102,6 +105,7 @@ public void init(HttpServletRequest req) {
     protected void initialiseAttributes(HttpServletRequest req) {
         request = req;
         requestUrl = HttpRequestHelper.getRequestedUrl(request);
+        currentUrl = request.getHeader("referer");
```

Comment:
Why `currentUrl` when `referer` (or rather, `referrer`) can make a better variable name?

Reply:
That does sound better! Changing this to referrer.

Reply:
Can use the new variable in `validateOriginIfRequired`.